rfc-template.md

```diff
@@ -0,0 +1,73 @@
+# Title
+
+- Contribution Name: (what do you want to call your `awesome_feature`?)
+- Implementation Owner: (your github @username)
+- Start Date: (today's date, dd-mm-yyyy)
+- Target Date: (expected date of completion, dd-mm-yyyy)
+- Appwrite Issue:
+  [Is this RFC inspired by an issue in appwrite](https://github.com/appwrite/appwrite/issues/)
+
+## Summary
+
+[summary]: #summary
+
+Brief explanation of the proposed contribution
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+- What problem are you trying to solve?
+- What is the context or background in which this problem exists?
+- Once the proposal is implemented, how will the system change?
+
+Please avoid discussing your proposed solution.
+
+## Design proposal (Step 2)
```

Comment:
Missing a few pointers that we talked about, like naming conventions, api routes, breaking changes. We should encourage discussions on this points.